src/interfaces/python/swig_typemaps.i

```diff
@@ -1312,10 +1368,13 @@ def _internal_factory_wrapper(object_name, new_name, docstring=None):
         _internal_factory.__doc__ = docstring
     else:
         _internal_factory.__doc__ = _obj.__doc__.replace(object_name, new_name)
+    _internal_factory.__qualname__ = new_name
 
     return _internal_factory
 
-kernel = _internal_factory_wrapper("_kernel", "kernel")
+for factory in _FACTORIES:
+	_rename_python_function(sys.modules[__name__], factory[1], factory[0])
```

Comment:
I suggest to add some docs/comments at 3 places
One for overview of what this is doing (code is slightly cryptic), here.
Both functions that are called in the loop should have an interface doc (Python style or whatever) intended for devs looking at this in a few years time when all of us are long gone